python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -390,6 +412,30 @@ pyCreateNativeKernel(const std::string &name, MlirModule module,
     setStateFunc(reinterpret_cast<std::complex<float> *>(svdata.data));
   }
 
+  // If we have any cudaq state data, we need to extract the function pointer
+  // to set that data, and then set it.
+  for (auto &[stateHash, stateData] : *cudaqStateStorage) {
+    if (platform.is_remote() || platform.is_emulated())
+      throw std::runtime_error("captured vectors are not supported on quantum "
+                               "hardware or remote simulators");
+
+    if (stateData.kernelName != name)
+      continue;
+
+    auto setStateFPtr = jit->lookup("nvqpp.set.cudaq.state." + stateHash);
+    if (auto error = setStateFPtr.takeError()) {
+      auto message = "python alt_launch_kernel failed to get set cudaq state "
+                     "function for kernel: " +
+                     name;
+      llvm::logAllUnhandledErrors(std::move(error), llvm::errs(), message);
```

Comment:
Why is this one using the `logAllUnhandledErrors` function but other cases throw a runtime_exception?

Reply:
I moved the changes not directly related to enabling new launch kernel in python to: https://github.com/NVIDIA/cuda-quantum/pull/2780